gsy_framework/sim_results/cumulative_grid_trades.py

```diff
@@ -192,11 +192,11 @@ def _update_area_children_in_accumulated_trades_dict(self, accumulated_trades, a
         Returns: None
         """
         current_child_uuids = [
-            c["uuid"] for c in accumulated_trades[area["uuid"]]["children"]
+            c["uuid"] for c in accumulated_trades[area["uuid"]].get("children", [])
```

Comment:
Can you please explain, what you are trying to achieve here? 
If `children=None`, then the for loop will still fail.

Reply:
I think that if `children = None`, there would be an exception. I suppose that's why Kamil is trying to replace the `None` with a `[]`.

Reply:
But this is not the way to replace `None` with `[]`:
```python
di = {"name": "test", "children": None}
print(di.get("children", []))
# None
```

Reply:
Sorry, my answer was incomplete/wrong: `.get()` will help in case `children` is not present at all. If `children` is `None`, then you need something like `.get("children") or []`.

Reply:
I expected that children will be missing or it will be a list.  e.g. this [line](https://github.com/gridsingularity/gsy-framework/blob/8abdd3fc44527cd5aebccb621feeeba6b38ffe6a/gsy_framework/sim_results/cumulative_grid_trades.py#L229) is creating the entries in the accumulated_trades so there will be no None value for children. This [line](https://github.com/gridsingularity/gsy-framework/blob/8abdd3fc44527cd5aebccb621feeeba6b38ffe6a/gsy_framework/sim_results/cumulative_grid_trades.py#L180) is creating a child dict and there is an option for missing children key. Although, you are right, if somehow there will be children=None, then this is not a fix. I can add it.

Reply:
The main error was:
```
File "/usr/local/lib/python3.8/site-packages/gsy_framework/sim_results/cumulative_grid_trades.py", line 195, in _update_area_children_in_accumulated_trades_dict
c["uuid"] for c in accumulated_trades[area["uuid"]]["children"]
KeyError: 'children'
```
which probably points to the missing `children` key, and not children: None

Reply:
But is this area supposed to have children? I mean assets do not have children per defenition.

Reply:
I didn't get to the root cause so I don't know for sure. What I suspect is somehow the area in the accumulated_trades was overwritten and lost their children. Because the other way, the leaf node (area without children) will never reach the point of `_accumulate_area_trades`.

Reply:
Is this area supposed to have children? [_accumulate_area_trades](https://github.com/gridsingularity/gsy-framework/blob/8abdd3fc44527cd5aebccb621feeeba6b38ffe6a/gsy_framework/sim_results/cumulative_grid_trades.py#L87) is reached only by Area type, so I think that it is usually should have children.

Reply:
I see. Indeed, then this is only a quick fix and probably not the fix of the root cause. I can also give it a try when you can not find it. Please let me know.